visual-diff.js

```
@@ -50,6 +51,18 @@ class VisualDiff {
 		}
 		_testNames.push(name);
 
+		if (dir) {
+			// "dir" can be either "__dirname" from a non-ESM source,
+			// or "import.meta.url" from an ESM source.
+			try {
+				const parentDirAsURL = new URL('.', dir);
+				dir = url.fileURLToPath(parentDirAsURL);
+			// eslint-disable-next-line no-empty
+			} catch (e) {}
```

When will this throw?  Do we want to set `dir` to something in that case?

---

It throws when `dir` isn't a URL -- i.e. in the existing case where they're passing in `__dirname` and it's a full path. So in that case we want to swallow the exception and just use `dir` as it was.

---

Once we convert this to ESM we can remove all of it and treat it as always being a URL.